src/magic_folder/config.py

```diff
@@ -1123,6 +1124,32 @@ def get_currentsnapshot_pathstate(self, cursor, name):
                 return PathState(mtime_ns=row[0], ctime_ns=row[1], size=row[2])
             raise KeyError(name)
 
+    @with_cursor
+    def get_all_current_snapshot_pathstates(self, cursor):
+        """
+        Return the PathState for every file we have in our
+        'current_snapshot' table.
+
+        :returns Iterable[(unicode, PathState)]: an iterable of
+            2-tuples of (name, PathState instance), one for each file
+        """
+        cursor.execute(
+            """
+            SELECT
+                name, mtime_ns, ctime_ns, size
+            FROM
+                current_snapshots
+            ORDER BY
+                mtime_ns DESC
+            """,
+        )
+
+        for row in cursor.fetchall():
+            yield (
```

Comment:
Don't `yield` in a function decorated by `with_cursor`, it doesn't do what you expect (the transaction is closed before any code in the function runs).

In any case, I don't think we should be holding open a transaction while the generator runs. In this case, it isn't an issue, since it consumed synchronously without other calls to `MagicFolderConfig`, but it seems like the kind of thing that will cause unexpected surprises down the line.

Reply:
I added a check to #470 that will detect and error in this case.